phonebook.py

```diff
@@ -0,0 +1,83 @@
+import json
+import re
+
+# ============================================================
+# Переносимо телефонну книгу з python-файлу в новий json-файл:
+phonebook = {
+    'Anna':'+111111111',
+    'Bogdan':'+222222222',
+    'Vova':'+333333333',
+    'Gena':'+444444444'
+}
+
+json_phonebook = json.dumps(phonebook)
+
+try:
+    with open("phonebook.json", 'x') as f:
+        f.write(json_phonebook)
+except FileExistsError:
+    print('File "phonebook.json" already exists')
+
+# ============================================================
+# Завантажуємо дані з існуючого json-файлу для роботи з ними:
+with open("phonebook.json", 'r') as f:
+    json_phonebook = f.read()
+
+phonebook = json.loads(json_phonebook)
+print(phonebook)
+
+command = ''
+while command != 'exit':
+    command = input("Enter a command stats/add/delete/list/show or exit: ")
+
+    match command:
+        case 'stats': # stats: кількість записів
+            key_list = phonebook.keys()
+            print(len(key_list))
+
+        case 'add': # add: додати запис
+            name = input("Enter a name: ")
+
+            if name in phonebook:
+                print("Contact with this name already exists.")
+            else:
+                while True:
```

Comment:
If you want to use three separate patters and check them separately it is not necessary to put them in the infinite loop or use such a big construction of `if .. elif ... else`. You can simplify your code like this (example where the infinite loop is still using):
```
if any([re.fullmatch(r'\+380\d{9}$', phonenumber), re.fullmatch(r'380\d{9}(?=$)', phonenumber), re.fullmatch(r'0\d{9}\b', phonenumber)]):
    break
else:
    print("Invalid format of a phone number. Try again.")

Reply:
Created one pattern for all cases